include/swift/SIL/SILBuilder.h

```diff
@@ -782,7 +782,7 @@ class SILBuilder {
   }
 
   void emitEndBorrowOperation(SILLocation loc, SILValue v) {
-    if (!hasOwnership())
+    if (!hasOwnership() || v.getOwnershipKind() == OwnershipKind::None)
```

Comment:
Should we have the same check we have in emitBeinBorrowOperation instead ?
`
   if (!hasOwnership() ||
        v.getOwnershipKind().isCompatibleWith(OwnershipKind::Guaranteed))
      return v;
`

Reply:
No. We want to be able to take guaranteed parameters here to end lifetimes.